### DISCUSSION TITLE
Velocities normalization in -coord=spherical_polar -s or -coord=schwarzschild -g

### DISCUSSION THREAD

**Discussion by LuisFLongo** [PrincetonUniversity/athena#554] [Q&A]
Hello everyone,

I am trying to pass data from a given part simulation of mine to Athena to push the  simulation further.
In this scenario I have to pass (among other quantities) the velocity-field of the fluid.

My question is which velocity I should pass in the following cases : 

-coord=spherical_polar -s 
                  or                   
-coord=schwarzschild   -g  

More specifically, it is my understanding that the primitive velocities in Athena (u) are defined as
u = Wv , where W is the Lorentz factor (as per above eq 31 in https://arxiv.org/pdf/2005.06651.pdf ),
but what is the exact definition of v (more specifically in the angular coordinates)?

meaning

$$v_{\theta} = \dfrac{d \theta }{d \tau}, \rm{implying   }   [v_{\theta}] = (rad/s) $$

or

$$v_{\theta} = j \dfrac{d \theta }{d \tau}, \rm{implying   }   [ v_{\theta} ] = (m/s) $$

where j comes from the Jacobian transformation?

**Comment by c-white** [comment]
In Athena, SR is made to be as close as possible to standard Newtonian physics, and so in particular all velocities have units of length/time. We are in an orthonormal (hatted) basis, rather than a natural/coordinate (unhatted) basis. The only difference is that in SR we use the spatial components of 4-velocity, which has better numerical properties than the 3-velocity. That is, the primitive velocities are $u^{\hat{r}} = u^r = \mathrm{d}r / \mathrm{d}\tau$, $u^{\hat{\theta}} = r u^\theta = r \mathrm{d}\theta / \mathrm{d}\tau$, and $u^{\hat{\phi}} = r \sin(\theta) u^\phi = r \sin(\theta) \mathrm{d}\phi / \mathrm{d}\tau$.

If you want to think of $u^{\hat{\imath}} = \gamma v^{\hat{\imath}}$, with $\gamma$ (equivalently $W$, equivalently $u^t = \mathrm{d}t / \mathrm{d}\tau$) the Lorentz factor, then $v^{\hat{r}} = \mathrm{d}r / \mathrm{d}t$, $v^{\hat{\theta}} = r \mathrm{d}\theta / \mathrm{d}t$, and $v^{\hat{\phi}} = r \sin(\theta) \mathrm{d}\phi / \mathrm{d}t$. And of course $\gamma = (1 - v^{\hat{r}} v^{\hat{r}} - v^{\hat{\theta}} v^{\hat{\theta}} - v^{\hat{\phi}} v^{\hat{\phi}})^{-1/2} = (1 + u^{\hat{r}} u^{\hat{r}} + u^{\hat{\theta}} u^{\hat{\theta}} + u^{\hat{\phi}} u^{\hat{\phi}})^{1/2}$. (The argument to the square root in the last expression being at least 1 no matter the primitive velocities is why we use 4-velocities rather than 3-velocities.)

As soon as you go to GR, Athena switches to a coordinate basis for all vectors, since doing anything else becomes confusing and pointless in the presence of non-diagonal metrics. The primitive velocities are still the spatial components of a 4-velocity, but now in a natural basis. Also, this is the normal observer (primed) frame, not the coordinate (unprimed) frame, but the two agree on spatial components of vectors in metrics like Schwarzschild with no shift. Thus the primitives are $u^{r'}$, $u^{\theta'}$, and $u^{\phi'}$, which have units of length/time, angle/time, and angle/time. In Schwarzschild coordinates, these happen to be equal to $u^r = \mathrm{d}r / \mathrm{d}\tau$, $u^\theta = \mathrm{d}\theta / \mathrm{d}\tau$, and $u^\phi = \mathrm{d}\phi / \mathrm{d}\tau$.

For completeness, the normal-frame Lorentz factor is $u^{t'} = (1 + g_{ij} u^{i'} u^{j'})^{1/2}$ (again, the square root is of a number no smaller than 1, no matter the primitives, which is why we use the normal frame in GR). The coordinate-frame Lorentz factor is $u^t = \alpha^{-1} u^{t'} = (-g^{tt})^{1/2} u^{t'}$. The spatial components shift between coordinate and normal frames via $u^{i'} = u^i + \beta^i u^t = u^i - (g^{ti} / g^{tt}) u^t$.

**Reply by LuisFLongo** [reply]
By $$\tau$$ there, do you mean the proper time, as in $$d \tau = dt / \gamma$$ in SR, and $$d \tau = \alpha dt $$ for GR ?

**Reply by c-white** [reply]
In SR, yes. In GR, it should be essentially the same thing: $\mathrm{d}\tau = \mathrm{d}t / u^t$. $\alpha$ connects the two different times, $\mathrm{d}t' = \alpha \mathrm{d}t$.

**Reply by LuisFLongo** [reply]
@c-white 
Thanks very much for the very complete and prompt answer!
